ReSharper.FSharp/src/FSharp/FSharp.Psi.Features/src/AICore/FSharpFileSummarizer.fs

```diff
@@ -1,36 +1,271 @@
 namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.AICore
 
+open System
+open System.Collections.Generic
+open System.Text
 open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Util.FcsTypeUtil
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
 open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Util
 open JetBrains.ReSharper.Psi
+open JetBrains.ReSharper.Psi.Tree
 open JetBrains.Rider.Backend.Features.AICore.Summarization
 open JetBrains.ReSharper.Psi.Files
+open FSharp.Compiler.Symbols
+open JetBrains.Util
+
+type private SummarizerContext() =
+    let scopes =  Stack<string>()
+    let summary = StringBuilder()
+
+    member x.CurrentScope = scopes.TryPeek()
+    member x.Summary = summary.ToString()
+
+    member x.AddEntity(node: ITreeNode, representation: string) =
```

Comment:
Is it possible to use any more specific type than `ITreeNode`? It's going to be difficult to maintain this code without proper types.

Reply:
In fact, we just need to get the `DocumentRange` that any `ITreeNode` has.